src/Products/Viewer/Controllers/ViewerApiController.cs

```diff
@@ -590,128 +689,139 @@ private static LoadDocumentEntity GetLoadDocumentEntity(GroupDocs.Viewer.Viewer
+        /// Calculates new page rotation angle value.
+        /// </summary>
+        /// <param name="currentAngle">Current page rotation angle value.</param>
+        /// <param name="postedAngle">Posted page rotation angle value.</param>
+        /// <returns>New page rotation angle value.</returns>
+        private static int GetNewAngleValue(int currentAngle, int postedAngle)
+        {
+            switch (currentAngle)
             {
-                //Do not release page stream as we'll need to keep the stream open
+                case 0:
+                    return postedAngle == 90 ? 90 : 270;
+                case 90:
+                    return postedAngle == 90 ? 180 : 0;
+                case 180:
+                    return postedAngle == 90 ? 270 : 90;
+                case 270:
+                    return postedAngle == 90 ? 0 : 180;
+                default:
+                    return 0;
             }
         }
 
-        private static PageDescriptionEntity GetPageDescriptionEntities(Page page, string pagesInfoPath)
+        /// <summary>
+        /// Gets enumeration member by rotation angle value.
+        /// </summary>
+        /// <param name="newAngle">New rotation angle value.</param>
+        /// <returns>Rotation enumeration member.</returns>
+        private static Rotation GetRotationByAngle(int newAngle)
         {
-            int currentAngle = GetCurrentAngle(page.Number, pagesInfoPath);
-
-            PageDescriptionEntity pageDescriptionEntity = new PageDescriptionEntity();
-            pageDescriptionEntity.number = page.Number;
-            // we intentionally use the 0 here because we plan to rotate only the page background using height/width
-            pageDescriptionEntity.angle = 0;
-            pageDescriptionEntity.height = currentAngle == 0 || currentAngle == 180 ? page.Height : page.Width;
-            pageDescriptionEntity.width = currentAngle == 0 || currentAngle == 180 ? page.Width : page.Height;
-            return pageDescriptionEntity;
+            switch (newAngle)
+            {
+                case 90:
+                    return Rotation.On90Degree;
+                case 180:
+                    return Rotation.On180Degree;
+                case 270:
+                    return Rotation.On270Degree;
+                default:
+                    return Rotation.On90Degree;
+            }
         }
 
-        private static string GetPageContent(Page page, string password, string documentGuid, ViewerSettings settings)
+        /// <summary>
+        /// Gets document pages data, dimensions and rotation angles.
+        /// </summary>
+        /// <param name="postedData">Posted data with document guid.</param>
+        /// <param name="loadAllPages">Flag to load all pages.</param>
+        /// <returns>Document pages data, dimensions and rotation angles.</returns>
+        private LoadDocumentEntity GetDocumentPages(PostedDataEntity postedData, bool loadAllPages)
```

Comment:
Codacy found an issue: [Make 'GetDocumentPages' a static method.](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5850423)